### DISCUSSION TITLE
Does YARP have a feature for marking a cluster’s destination as “offline/suspended”?

### DISCUSSION THREAD

**Discussion by jessecavada71** [dotnet/yarp#545] [Q&A] [answered]
Does YARP have a feature for marking a cluster’s destination as “offline/suspended”? The intent is to place a cluster’s destination in “offline/suspended” allowing any pending http responses to complete but not allowing any new http requests to the “offline/suspended” destination. The use case is when it is known that the cluster’s destination will be unavailable such as during deployment of a cluster’s destination or any other scheduled maintenance of the cluster’s destination.

**Comment by Tratcher** [comment, marked as answer]
Right now the recommendation would be to remove that destination from the config. The config will live update and stop sending requests to that destination, but will still allow active requests to complete.

See https://github.com/microsoft/reverse-proxy/issues/126 for discussions on a service rollout pattern.
